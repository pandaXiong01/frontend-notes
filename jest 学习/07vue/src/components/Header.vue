<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-inputs"
        v-model="inputValue"
        @keyup.enter="addTodoItem"
        placeholder="TodoItem"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data(){
    return {
      inputValue:''
    }
  },
  methods: {
    addTodoItem(){
      if(this.inputValue){
        this.$emit('add',this.inputValue)
        this.inputValue = ''
      }
    }
  },
}
</script>
<style scoped lang="scss">
.header {
  line-height: 60px;
  background: #333;
  .header-content {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
    .header-inputs {
      float: right;
      width: 360px;
      margin-top: 16px;
      line-height: 24px;
      outline: none;
      color: #333;
    }
  }
}
</style>
