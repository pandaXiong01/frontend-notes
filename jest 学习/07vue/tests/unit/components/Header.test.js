import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Header.vue'

describe('Header组件测试', () => {
  it('样式发生改变，做提示', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
    // const input = wrapper.find('[data-test="input"]')
    // expect(input.exists()).toBe(true)
  })

  it('input 框初始内容为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe('')
  })

  it('input 框发生变化，数据应该跟着改变', () => {
    const content = "zuobaiquan";
    const wrapper = shallowMount(Header)
    const input = wrapper.findAll(".header-inputs")
    input.setValue(content)
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe(content)
  })

  it('input 框回车，无内容时无反应', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.findAll(".header-inputs")
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('input 框回车，有内容时向外触发事件,同时清空 inputValue', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.findAll(".header-inputs")
    input.setValue('zuobaiquan')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.inputValue).toBe('')
  })
})
