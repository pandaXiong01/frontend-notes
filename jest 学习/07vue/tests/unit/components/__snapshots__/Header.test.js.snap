// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header组件测试 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input placeholder="TodoItem" class="header-inputs">
  </div>
</div>
`;
